buildSrc/src/main/groovy/org/elasticsearch/gradle/test/ClusterFormationTasks.groovy

```diff
@@ -356,14 +356,10 @@ class ClusterFormationTasks {
                 'node.attr.testattr'           : 'test'
         ]
         int minimumMasterNodes = node.config.minimumMasterNodes.call()
-        if (minimumMasterNodes > 0) {
+        if (node.nodeVersion.before("7.0.0") && minimumMasterNodes > 0) {
             esConfig['discovery.zen.minimum_master_nodes'] = minimumMasterNodes
         }
-        if (minimumMasterNodes > 1) {
-            // don't wait for state.. just start up quickly
-            // this will also allow new and old nodes in the BWC case to become the master
-            esConfig['discovery.initial_state_timeout'] = '0s'
-        }
+        esConfig['discovery.initial_state_timeout'] = '0s'
         if (esConfig.containsKey('discovery.zen.master_election.wait_for_joins_timeout') == false) {
```

Comment:
only do this for `node.nodeVersion.before("7.0.0")`

Reply:
Ok - see 7c8a540335a79cc8bad59c9621e37d904b1f336e. Note that I haven't been through all the other Zen1-specific settings yet, this PR is only really about `minimum_master_nodes`.